components/shared/base/id.rs

```diff
@@ -421,6 +422,19 @@ impl HistoryStateId {
     }
 }
 
+namespace_id! {GamepadRouterId, GamepadRouterIndex, "GamepadRouter"}
+
+impl GamepadRouterId {
```

Comment:
This is also unnecessary(but thanks for looking into it), because you don't need a router at the constellation level, just an IPC route will do.